honeybee/model.py

```diff
@@ -119,243 +207,391 @@ def north_vector(self, value):
+        return child_apertures + self._apertures
 
     @property
     def doors(self):
         """A list of all Door objects in the model."""
-        return self._doors
+        child_doors = []
+        for room in self._rooms:
+            for face in room._faces:
+                child_doors.extend(face._doors)
+        for face in self._faces:
+            child_doors.extend(face._doors)
+        return child_doors + self._doors
 
     @property
-    def has_ngon_apertures(self):
-        """Boolean noting whether the model has apertures with more than 4 sides.
-
-        This is helpful for energy simulation since EnergyPlus cannot accept
-        sub-faces with more than 4 sides. This property should be checked before
-        running the triangulate_ngon_apertures() method since many models do not
-        have such apertures.
-        """
-        for ap in self.apertures:
-            if len(ap.geometry) > 4:
-                return True
-        return False
+    def orphaned_faces(self):
+        """A list of all Face objects without parent Rooms in the model."""
+        return tuple(self._faces)
 
     @property
-    def has_ngon_doors(self):
-        """Boolean noting whether the model has doors with more than 4 sides.
-
-        This is helpful for energy simulation since EnergyPlus cannot accept
-        sub-faces with more than 4 sides. This property should be checked before
-        running the triangulate_ngon_doors() method since many models do not have
-        such doors.
-        """
-        for dr in self.doors:
-            if len(dr.geometry) > 4:
-                return True
-        return False
-
-    def get_room(self, name):
-        """Get a room object in the model given the room name."""
-        return self._rooms[self._room_names.index(name)]
-
-    def get_face(self, name):
-        """Get a face object in the model given the face name."""
-        return self._faces[self._face_names.index(name)]
+    def orphaned_shades(self):
+        """A list of all Shade objects without parent Rooms in the model."""
+        return tuple(self._shades)
 
-    def get_shade(self, name):
-        """Get a shade object in the model given the shade name."""
-        return self._shades[self._shade_names.index(name)]
-
-    def get_aperture(self, name):
-        """Get an aperture object in the model given the aperture name."""
-        return self._apertures[self._aperture_names.index(name)]
+    @property
+    def orphaned_apertures(self):
+        """A list of all Aperture objects without parent Faces in the model."""
+        return tuple(self._apertures)
 
-    def get_door(self, name):
-        """Get an door object in the model given the door name."""
-        return self._doors[self._door_names.index(name)]
+    @property
+    def orphaned_doors(self):
+        """A list of all Door objects without parent Faces in the model."""
+        return tuple(self._doors)
 
     def add_model(self, other_model):
```

Comment:
Very nice! should we also add `__add__` magic method and use this method under the hood so we can do: `model = model_1 + model_2`

Reply:
That does seem very elegant-looking.  I'll add in an `__add__`!